azurerm/internal/services/mssql/resource_arm_mssql_elasticpool.go

```diff
@@ -151,6 +151,16 @@ func resourceArmMsSqlElasticPool() *schema.Resource {
 				Optional: true,
 			},
 
+			"license_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.BasePrice),
+					string(sql.LicenseIncluded),
+				}, false),
+			},
```

Comment:
I did this to stay consistent I followed the code implemented in `resource_arm_mssql_database.go`.  When I first saw that in the schema, I was unsure to the effect.  My conclusion is it allowed you to avoid specifying a default.  I figured this was better behavior because you allowed upstream to maintain the default and prevents a bad guess on the downstream by developer. But I think it speaks more to the 2 item in your list.

1.  My undestanding is license_type would not change, unless the sku changed.  Which I think speak to your item 2 in the list.
2. I am not an expert with the resource.  The little bit of reading and experience I had with writing the tests, I found you could set the value of licenseType but if the sku.name was BasicPool the licenseType came back empty.  There is no doubt we could enhance the schema to only allow the field when it is relevant to sku.  But if we are requiring this, we should probably pull in a change for `resource_arm_mssql_database.go`.

I am up for what the maintainers recommend.

Reply:
Thanks @pearcec . I think Computed should be fine after a discussion with kt. Sorry for the inconvenience here

Reply:
@ArcturusZhang no problem it was good discussion.